problem with fetchings some test jobs

```
Sep 01 09:57:53 qa-reports squad-admin[28499]: [2017-09-01 09:57:53,755: INFO/Worker-1] squad.ci.tasks.fetch[5208cf97-8c5e-4839-bbe1-fcda38c6d9f2]: fetching lkft.validation.linaro.org/21984
Sep 01 09:58:55 qa-reports squad-admin[28499]: [2017-09-01 09:58:55 +0000] [ERROR] Task squad.ci.tasks.fetch[5208cf97-8c5e-4839-bbe1-fcda38c6d9f2] raised unexpected: Error()
Sep 01 09:58:55 qa-reports squad-admin[28499]: Traceback (most recent call last):
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/celery/app/trace.py", line 240, in trace_task
Sep 01 09:58:55 qa-reports squad-admin[28499]:     R = retval = fun(*args, **kwargs)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/celery/app/trace.py", line 438, in __protected_call__
Sep 01 09:58:55 qa-reports squad-admin[28499]:     return self.run(*args, **kwargs)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/squad/ci/tasks.py", line 27, in fetch
Sep 01 09:58:55 qa-reports squad-admin[28499]:     test_job.backend.fetch(test_job)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/squad/ci/models.py", line 48, in fetch
Sep 01 09:58:55 qa-reports squad-admin[28499]:     self.really_fetch(test_job)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/squad/ci/models.py", line 52, in really_fetch
Sep 01 09:58:55 qa-reports squad-admin[28499]:     results = implementation.fetch(test_job)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/squad/ci/backend/lava.py", line 45, in fetch
Sep 01 09:58:55 qa-reports squad-admin[28499]:     yamldata = self.__get_testjob_results_yaml__(test_job.job_id)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/srv/qa-reports.linaro.org/lib/python3.5/site-packages/squad/ci/backend/lava.py", line 180, in __get_testjob_results_yaml__
Sep 01 09:58:55 qa-reports squad-admin[28499]:     return self.proxy.results.get_testjob_results_yaml(job_id)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/usr/lib/python3.5/xmlrpc/client.py", line 1092, in __call__
Sep 01 09:58:55 qa-reports squad-admin[28499]:     return self.__send(self.__name, args)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/usr/lib/python3.5/xmlrpc/client.py", line 1432, in __request
Sep 01 09:58:55 qa-reports squad-admin[28499]:     verbose=self.__verbose
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/usr/lib/python3.5/xmlrpc/client.py", line 1134, in request
Sep 01 09:58:55 qa-reports squad-admin[28499]:     return self.single_request(host, handler, request_body, verbose)
Sep 01 09:58:55 qa-reports squad-admin[28499]:   File "/usr/lib/python3.5/xmlrpc/client.py", line 1167, in single_request
Sep 01 09:58:55 qa-reports squad-admin[28499]:     dict(resp.getheaders())
Sep 01 09:58:55 qa-reports squad-admin[28499]: xmlrpc.client.Error: Error()
```

--- a/squad/ci/backend/lava.py
+++ b/squad/ci/backend/lava.py
@@ -42,8 +42,16 @@ class Backend(BaseBackend):
         data = self.__get_job_details__(test_job.job_id)
 
         if data['status'] in self.complete_statuses:
-            yamldata = self.__get_testjob_results_yaml__(test_job.job_id)
-            data['results'] = yaml.load(yamldata)
+            try:
+                yamldata = self.__get_testjob_results_yaml__(test_job.job_id)
+                data['results'] = yaml.load(yamldata)
+            except xmlrpclib.Error as e:
+                self.log_error("Error during result fetching")
+                self.log_error(str(e) + "\n" + traceback.format_exc())
+                test_job.failure = str(e) + "\n" + traceback.format_exc()
+                test_job.save()
+                send_admin_email.delay(test_job.pk)
+                return None
 
             # fetch logs
             logs = ""

